\documentclass{TMarticle}

\usepackage[]{lipsum}
\usepackage[]{paralist}
\usepackage[]{amsmath}
\usepackage[]{amssymb}
\definecolor{TMcodeBackground}{RGB}{240, 240, 240}
\definecolor{TMbulletinBackground}{RGB}{240, 240, 240}
\author{Danilo Horta}

\title{Generalised linear mixed models for genetic analysis}

\begin{document}

\maketitle

\section{abstract}

The development of computationally efficient yet accurate models has received
considerable attention in statistical genetics. In particular linear mixed
models (LMMs) are now a well established tool and provide powerful control for
population structure and relatedness, allow to aggregate across multiple causal
variants in gene sets and can be used to leverage phenotype correlations between
multiple (related) traits.

However, the vast majority of existing LMM approaches assume that phenotypes
are continuous with Gaussian distributed residuals. This assumption is clearly
violated in case/control studies but also in the context of a many
sequencing-based phenotypes, such as Poisson distributed read count data or
traits defined as the Binomial ratio of (typically small) count values. While
generalised linear mixed models provide in principle an established solution to
this problem, "exact" methods for parameter inference require expensive MCMC
simulations and hence are not applicable to large cohorts. Consequently,
non-Gaussian observation likelihood are in practice either ignored or one is
left with methods that provide crude approximations to estimate the trait on a
latent liability scale.

To address this, we here propose a highly effective deterministic algorithm
QEP-LMM that enables near-exact marginalising over the latent liability scale
within the LMM framework. This model provides quadratic and in some instances
even linear run-time complexity in the number of samples, thus enabling the
analysis of datasets with tens of thousands of individuals in the context of
genome-wide tests. We extensively compared our model with existing
state-of-the-art tools (Gaussian LMM, GCTA, LTMLM, MACAU, and LEAP), both in
terms of power to detect associations as well as accuracy for heritability
estimation, phenotype prediction, and computational performance. Consistently
across settings, we find substantial improvements over current approximate
methods. Remarkably, we observe that QEP-LMM achieves near-identical performance
to exact MCMC approaches for generalised LMMs at a runtime complexity that is
comparable to a standard LMM. Finally, we provide empirical results to
demonstrate practical utility of QEP-LMM in applications to data from the WTCCC
and in the genetic analysis of splicing phenotypes in human LCLs.

## Bibliography

FALCONER, D.S., 1965. The inheritance of liability to certain diseases, estimated from the incidence among relatives. Annals of Human Genetics, 29(1), pp.51–76. Available at: http://onlinelibrary.wiley.com/doi/10.1111/j.1469-1809.1965.tb00500.x/abstract.

Lee, S.H. et al., 2011. Estimating Missing Heritability for Disease from Genome-wide Association Studies. The American Journal of Human Genetics, 88(3), pp.294–305. Available at: http://www.cell.com/ajhg/abstract/S0002-9297(11)00020-6.

McCullagh, P. & Nelder, J. A. (1989), Generalized Linear Models , Chapman & Hall / CRC , London.

## Appendix

### Probability distributions

| name        | pmf/pdf                                  | parameters                           | support                    | mean            | variance          |
| ----------- | ---------------------------------------- | ------------------------------------ | -------------------------- | --------------- | ----------------- |
| Binomial    | ${n \choose ny} p^{ny}(1-p)^{n-ny}$      | $n\in \{1, 2, \dots\}\\p\in[0,1]$    | $y=\{0/n, 1/n, \dots, 1\}$ | $p$             | $p(1-p)/n$        |
| Bernoulli   | set $n=1$                                |                                      |                            |                 |                   |
| Gamma       | $\frac{(ay)^{a-1} e^{-\frac{ay}{s}} a}{\Gamma(a)s^a}$ | $a>0\text{ shape}\\s>0 \text{ rate}$ | $y \in [0, +\infty)$       | $s$             | $s^2/a$           |
| Exponential | set $a=1$                                |                                      |                            | $1$             | $1/s$             |
| Poisson     | $\frac{\lambda^y e^{-\lambda}}{y!}$      | $\lambda > 0$                        | $y \in \{0, 1, \dots\}$    | $\lambda$       | $\lambda$         |
| Geometric   | $(1-p)^y p$                              | $p\in[0,1]$                          | $y \in \{0, 1, \dots\}$    | $\frac{1-p}{p}$ | $\frac{1-p}{p^2}$ |

### Probability distribution in natural form

| name        | pmf/pdf                                  | theta                   | phi  | a(phi)   |      |
| ----------- | ---------------------------------------- | ----------------------- | ---- | -------- | ---- |
| Binomial    | $\exp\left\{(y\theta - \log(1+e^\theta))/n^{-1} + \log {n \choose yn}\right\}$ | $\log\{\frac{p}{1-p}\}$ | $n$  | $1/\phi$ |      |
| Bernoulli   |                                          |                         |      |          |      |
| Gamma       | $\exp\left\{(y\theta - \log(-\theta^{-1}))/a^{-1}\right\} + a\log(a) + (a-1)\log(y) - \log\Gamma(a)\}$ | $-1/s$                  | $a$  | $1/\phi$ |      |
| Exponential |                                          |                         |      |          |      |
| Poisson     | $\exp\left\{(y\theta - e^\theta) - \log(y!)\right\}$ | $\log(\lambda)$         | $1$  |          |      |
| Geometric   | $\exp\left\{y\theta + \log(1-e^\theta)\right\}$ | $\log(1-p)$             | $1$  |          |      |

### Info I don't want to forget

"For binary traits, such as disease, familial resemblance is usually parameterized on an unobserved continuous liability scale so that the heritability is independent of disease prevalence. (FALCONER 1965; Lee et al. 2011)"

"For complex diseases it would be very useful to apply the same estimation procedure to case-control GWAS data. However, there are three issues that need to be overcome to be able to estimate genetic variance for disease without bias and with computationally fast algorithms:

Quality control (QC) of SNPs. QC is more of a concern for case-control than quantitative GWAS. For quantitative traits, experimental or genotyping artifacts are unlikely to be correlated with the trait value. However, case and control sets are often collected independently so that experimental artifacts could make cases more similar to other cases and controls more similar to other controls. These artificial case-control diferences could be partitioned as 'heritability' in methods that utilize genome-wide similarity within and diferences between cases and controls."







![Screen Shot 2016-10-12 at 17.26.19](/Users/horta/Desktop/Screen Shot 2016-10-12 at 17.26.19.png)

![Screen Shot 2016-10-12 at 17.12.54](/Users/horta/Desktop/Screen Shot 2016-10-12 at 17.12.54.png)


\end{document}
